src/functionalTest/java/uk/gov/hmcts/darts/audio/AudioTransformationServiceFunctionalTest.java

```diff
@@ -0,0 +1,184 @@
+package uk.gov.hmcts.darts.audio;
+
+import io.restassured.builder.MultiPartSpecBuilder;
+import io.restassured.http.ContentType;
+import io.restassured.response.Response;
+import io.restassured.specification.MultiPartSpecification;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import uk.gov.hmcts.darts.FunctionalTest;
+import uk.gov.hmcts.darts.cases.model.AdvancedSearchResult;
+
+import java.io.IOException;
+
+import static org.apache.commons.lang3.RandomStringUtils.randomAlphanumeric;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+@SuppressWarnings({"VariableDeclarationUsageDistance", "CommentsIndentation"})
+class AudioTransformationServiceFunctionalTest extends FunctionalTest {
+
+    private static final String CASES_PATH = "/cases";
+    private static final String AUDIOS_PATH = "/audios";
+    private static final String CASE_SEARCH_URL = "/cases/search";
+    private static final String AUDIO_REQUESTS_PATH = "/audio-requests";
+
+
+    @AfterEach
+    void cleanData() {
+        buildRequestWithExternalAuth()
+            .baseUri(getUri("/functional-tests/clean"))
+            .redirects().follow(false)
+            .delete();
+    }
+
+
+    @Test
+    void testPostAudio() throws IOException {
+        String courthouseName = "func-swansea-house-" + randomAlphanumeric(7);
+        String courtroomName = "func-swansea-room-" + randomAlphanumeric(7);
+        String caseNumber = "func-case-" + randomAlphanumeric(7);
+
+        createCourtroomAndCourthouse(courthouseName, courtroomName);
+
+        createCase(courthouseName, caseNumber);
+
+        postAudio(courthouseName, courtroomName, caseNumber, "functional-test-ch1.mp2", 1);
+        postAudio(courthouseName, courtroomName, caseNumber, "functional-test-ch2.mp2", 2);
+        postAudio(courthouseName, courtroomName, caseNumber, "functional-test-ch3.mp2", 3);
+        postAudio(courthouseName, courtroomName, caseNumber, "functional-test-ch4.mp2", 4);
+
+        int hearingId = getHearingIdByCaseNumber(caseNumber);
+
+        assertNotNull(hearingId);
+
+        String audioRequest = """
+            {
+              "hearing_id": <<hearingId>>,
+              "requestor": 0,
+              "start_time": "2024-03-11T10:15:00.000Z",
+              "end_time": "2024-03-11T10:20:00.000Z",
+              "request_type": "DOWNLOAD"
+            }
+            """;
```

Comment:
I understand that one of the challenges with the functional test approach is that when it comes to run the test against staging (which happens upon merging into master), creating a media request may result in KEDA picking the request up and processing it before the test has a chance to.

Could a solution to that be:
- Create a new endpoint in `TestSupportController` that allows the func test to create a media request with a non-`OPEN` `status`. E.g. `PROCESSING`.
- Invoke `/functional-tests/handleKedaInvocationForMediaRequests/` with this requestId.

I think this would avoid the issue, as KEDA will only detect `OPEN` requests. The `processAudioRequest()` method doesn't care what the status is, it will overwite it to `PROCESSING` anyway.

A bit hacky, but might be an option to explore?

Reply:
This was the reason that this was not continued past this point as there was a potential race condition. Again its all got a bit hacky and then to actually do some verification of handleKedaInvocationForMediaRequests would require to either add a new functional test endpoint that uses the integration database tools to test the db is as expected or use current endpoints that may not give you all of what you want to verify.